src/domain/repositories/DocumentRepository.ts

```diff
@@ -0,0 +1,7 @@
+import { Document } from "../entities/Document";
+
+export interface DocumentRepository {
```

Comment:
Cool! It's great that we've gotten the logic on InstanceRepository for documents out of the way.